backend/readers_backend/core/views.py

```diff
@@ -54,6 +55,22 @@ def get_permissions(self):
         if self.action in ["list", "retrieve", "latest"]:
             return []
         return [IsAuthenticated()]
+    
+    def validate_query_params(self, params):
+        '''
+        Throws validation error if any of the query params are not valid.
+        '''
+        if params.get("major_version") is not None:
```

Comment:
This feels a lot like duplicating work the framework should be doing for you. Can you use a Serializer to perform the parsing and validation of the query params? https://forum.djangoproject.com/t/parsing-and-validating-query-params-from-get-request/24308/2